src/mozilla_taskgraph/worker_types.py

```diff
@@ -17,9 +17,32 @@
                 "env", description="Environment variables to pass into the build"
             ): {str: taskref_or_string},
             Optional(
-                "build_params",
-                description="Parameters describing the build context to pass "
-                "onto Bitrise. All keys are optional but specific workflows "
+                "workflow_params",
+                description="Workflow-specific parameters describing the build context to pass "
+                "onto Bitrise. The key should be the workflow id, and mapped to the params to be added. "
+                "Overrides global_params if there's a conflict. Lists are MERGED.",
+            ): {
+                str: {
```

Comment:
In your scriptworker-scripts PR I'm pretty sure this was a list of permutations, so the immediate value here should be a list.

That said, there might be other nicer ways of setting that up.